packages/amplify-codegen/src/commands/models.js

```diff
@@ -76,7 +76,12 @@ async function generateModels(context, outputDirPath = null, isIntrospection = f
   });
 
   const schemaContent = loadSchema(apiResourcePath);
-  const outputPath = outputDirPath || path.join(projectRoot, getModelOutputPath(context));
+
+  const outputDirParam = context.parameters?.options?.['output-dir'];
+  if ( !outputDirPath && outputDirParam && typeof(outputDirParam) !== 'string' ) {
+    throw new Error('Expected provided --output-dir flag to be given output location as input.');
+  }
```

Comment:
@AaronZyLee - do you think we need this validation here? This seems like it'd still work if we just yanked lines 81-83

Reply:
Actually, I suppose it doesn't hurt anything in particular.

Reply:
I think the validation is for the case where we provide `--output-dir` without a value, which is set as `true`. The following command should be invalid.
```
$ amplify codegen models --output-dir
```